lib/ui_layer/side_drawer/search_tab.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:provider/provider.dart';
+import 'package:studentmap/domain_layer/building_manager/building.dart';
+import 'package:studentmap/domain_layer/building_manager/building_manager.dart';
+import 'package:studentmap/ui_layer/widgets/search_tab/search_result.dart';
+
+/// This tab allows the user to search for a building by name
+/// Takes a [TextEditingController] object
+
+class SearchTab extends StatefulWidget {
+  const SearchTab(this.controller, {super.key});
+
+  final TextEditingController controller;
+
+  @override
+  State createState() => _SearchTab(controller);
+}
+
+class _SearchTab extends State<SearchTab> {
+  _SearchTab(this.controller);
+
+  final TextEditingController controller;
+  List<Building> returnResults = [];
+
+  @override
+  Widget build(BuildContext context) {
+    BuildingManager bm = context.read<BuildingManager>();
+    return Scaffold(
+        body: Column(
+      children: <Widget>[
+        // Search Section
+        Row(
+          children: [
+            Flexible(
+              child: TextField(
+                controller: controller,
+                onChanged: (value) {
+                  returnResults = bm.getMatchingBuildingsByName(value, limit: 10);
+                  setState(() {});}
+              ),
+            ),
+            Padding(
+              padding: const EdgeInsets.only(
+                  left:
+                      8.0), // Add some space between the text field and button
+              child: ElevatedButton(
+                onPressed: () {
+                  FocusScope.of(context).unfocus();
+                },
+                child: const Text("Search"),
+              ),
+            ),
+          ],
+        ),
+        const Padding(
+            padding: EdgeInsets.symmetric(vertical: 30.0), child: Divider()),
+
+        //Search Results
+        Expanded(
+            child: ListView.builder(
+                itemCount: returnResults.isEmpty ? 1 : returnResults.length,
+                itemBuilder: (BuildContext context, int index) {
+
+                  if(returnResults.isEmpty) return const Text("   No Results"); 
```

Comment:
We should probably wrap Text inside of a Padding e.g. `Padding(padding: EdgeInsets.only(left: 4), child: Text())` instead of padding the string with spaces